src/main/scala/beam/agentsim/agents/PersonAgent.scala

```diff
@@ -833,6 +859,29 @@ class PersonAgent(
           } else {
             currentVehicle
           }
+        // todo rrp
+        currentVehicleForNextState.headOption.foreach { beamVehicleId =>
+          import beamScenario.beamConfig.beam.agentsim.agents.vehicles.{
+            noRechargeThresholdInMeters,
+            rechargeRequiredThresholdInMeters
+          }
+
+          val vehicle = beamVehicles(beamVehicleId).vehicle
+          if (vehicle.isBEV || vehicle.isPHEV) {
+            val distanceToDestinationInMeters = trip.foldLeft(0.0)(_ + _.beamLeg.travelPath.distanceInM)
```

Comment:
I think we need to filter the trip so that leg.vehicle == vehicle. The rest of trip can include the walk and/or transit legs.

Reply:
Ah, nice catch. Thanks! I'll add filter part. :+1: